### PR TITLE
Making Some Change

### PR DESCRIPTION
**Pull request by kacihmd**

### CODE REVIEW

**Comment by kacihmd** [comment]
Hurry up before my impatience pushes me to merge this bomb !

File: galamadriabuyak/ICharacter.java [hunk 1]
```diff
@@ -8,7 +8,8 @@
  * When a character's health points fall under zero, he dies.
  * @inv
  *      getName() != null
- *      getHealth() <= 0 ==> isDead()
+ *      getHealth() >= 0
```
Comment (durierem) [inline galamadriabuyak/ICharacter.java:11]: This is not an invariant. getHealth() can be lesser than 0.

File: galamadriabuyak/ICharacter.java [hunk 2]
```diff
@@ -8,7 +8,8 @@
...
+ *      getHealth() == 0 ==> isDead()
```
Comment (durierem) [inline galamadriabuyak/ICharacter.java:12]: I think getHealth() <= 0 would be better

File: galamadriabuyak/ICharacter.java [hunk 3]
```diff
@@ -97,6 +98,7 @@
      *      q >= 0
      * @post
      *      getHealth() == q
+     *      q == 0 => isDead()
```
Comment (durierem) [inline galamadriabuyak/ICharacter.java:101]: Same shit, q could be <= 0

File: galamadriabuyak/ICharacter.java [hunk 4]
```diff
@@ -114,7 +116,8 @@
      * @pre
      *      q >= 0
      * @post
-     *      getHealth() == old getHealth() - q
+     *      getHealth() == min(0, old getHealth() - q)
+     *      getHealth() == 0 ==> isDead()
```
Comment (durierem) [inline galamadriabuyak/ICharacter.java:120]: <= 0 again

**Review by durierem — APPROVED** [review]
Great change, superior creature!

**Pull request merged**